### PR TITLE
Allow chains of reshape

### PR DESCRIPTION
**Pull request by naoyam**
Previously, fusions like [this](https://github.com/NVIDIA/Fuser/pull/60/files#diff-a8f5333aa3f2d21440b3cea429bb2a588ed583f4d05486063ef1dc1a30996df9R2411) are segmented due to a limitation of `DomainMap`. 

It seems there's no impact to the existing tests and benchmarks. No failure with the tests and benchmarks. Dumped all CUDA generated kernels from the benchmarks and compared before and after this PR. Nothing changed.

This is part of the fix for #58

### CODE REVIEW

File: csrc/compute_at_map.h
Comment (naoyam) [inline csrc/compute_at_map.h:1]: Nothing changed except for `getAllDisjointSetProducers` and `getAllDisjointSetConsumers` are made available as public const functions.

File: csrc/scheduler/pointwise_utils.cpp
Comment (naoyam) [inline csrc/scheduler/pointwise_utils.cpp:1]: This file is the main change of this PR

Comment (zasdfgbnm) [inline csrc/scheduler/pointwise_utils.cpp:1]: `eraseIfMapped` uses permissive mapping, but here we are using exact map traverse in `eraseIfInputMappedThroughRFactorDomain`, so I feel worried about if this change will break some corner cases that we have never thought about, although I have no concrete example. But the benefit of this change is clearly larger than this potential risk, so we can just take this in and fix it later if there is any new bug.

**Review by zasdfgbnm — APPROVED** [review]

File: csrc/scheduler/pointwise_utils.cpp
Comment (naoyam) [reply csrc/scheduler/pointwise_utils.cpp:1]: I think this keeps the same behavior as before. Note that the exact map is only used when grabbing all mapped IDs. We still use `eraseIfMapped`, i.e., the permissive map, to determine if an IterDomain of an input tensor is considered mapped or not.

**Pull request merged**